### ISSUE TITLE
Update serialized references when refactoring method names

### ISSUE THREAD

**Issue by B-Reif** [JetBrains/resharper-unity#1300]
Method refactoring doesn't update Unity's serialized references to the target method.

When renaming a method, the developer must also manually update any of Unity's serialized references to that method (in e.g. `.anim` files). This creates a lot of toil. I would expect refactoring to update these.

**Comment by krasnotsvetov** [comment]
We support update in .prefab, .unity & .asset files. We are working on supporting .anim files too

**Comment by nick-morhun** [comment]
It seems that the new prefab variants system is not supported yet.

**Comment by SugoiDev** [comment]
Is this system extendable so users can add support for other serialized formats?

One important use-case is "visual scripting". There's quite a few those tools around for Unity and, due to  limitations in Unity's serialization system, they use their own serialization systems and formats. The most popular format being JSON.

Two serializers in particular that come to mind are FullSerializer (used by Slate and NodeCanvas and probably many others) and OdinSerializer (used at least by Odin Inspector).

**Comment by citizenmatt** [comment]
No, the plugin only supports Unity's YAML files, and doesn't cover all instances yet, either (e.g. `.anim` files).

Technically it's possible to extend this, in the same way that we're supporting YAML right now, but it would have to be written as a C# extension, and it's not exactly straightforward. If it would be worth adding support for FullSerializer and OdinSerializer file formats, then could you create a new issue with some details/repro steps, please?

**Comment by SugoiDev** [comment]
Alright. Create an issue for discussion and to gauge interest before I investigate it deeper.

**Comment by studentutu** [comment]
@citizenmatt  please see the issue I've described  - https://github.com/JetBrains/resharper-unity/issues/1918
